photutils/morphology.py

```diff
@@ -5,14 +5,27 @@
 """
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
+import collections
 import numpy as np
-from astropy.modeling import models
+from astropy.modeling.models import Gaussian1D, Gaussian2D, Const1D, Const2D
 from astropy.modeling.fitting import LevMarLSQFitter
+from astropy.nddata.utils import overlap_slices
 from .segmentation import SegmentProperties
+import warnings
 
 
-__all__ = ['centroid_com', 'gaussian1d_moments', 'centroid_1dg',
-           'centroid_2dg', 'fit_2dgaussian', 'data_properties']
+__all__ = ['GaussianConst1D', 'GaussianConst2D', 'centroid_com',
```

Comment:
In general, with the `..autoclass::` directive, you can add `:no-show-inheritance` to the directive to disable that.  Though I'm not sure how to work that in the context of the module autosummary.

Another thing you can do is add to nitpick_ignore in conf.py a la

https://github.com/astropy/astropy/blob/master/docs/conf.py#L193 and
https://github.com/astropy/astropy/blob/master/docs/nitpick-exceptions

I wonder if that supports patterns though.  Because really you want to ignore something like `photutils.morphology.CompoundModel*`.

A final possiblity is that I might might be able to just totally munge the `__bases__` for subclasses of generated compound models like the ones in this module.  I'll have to have a look at that sometime.